### PR TITLE
base_path parameter and relative pathes

### PR DESCRIPTION
**Pull request by NotJustPizza**
You can now specify base path for yml files:
```
app = Flask(__name__)

@app.route("/spec")
def spec():
    base_path = os.path.join(app.root_path, 'docs')
    return jsonify(swagger(app), from_file_keyword="swagger_from_file", base_path=base_path)
```
and use relative pathes:
```
@app.route('/test', methods=['POST'])
def login():
    """
    swagger_from_file: test.yml
    """
```

### CODE REVIEW

**Comment by NotJustPizza** [comment]
I added customizations made by @Steven-Eardley to this pull request

**Review by atlithorn — COMMENTED** [review]
My sincerest apologies. I somehow managed to turn off notifications on github and am first seeing this a year later when I stumbled inadvertently back to this project.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,24 @@
-# flask-swagger
+# flask-swagger (fork)
```
Comment (atlithorn) [inline README.md]: If you want this merged we will have to remove the "fork" messaging

File: flask_swagger.py [hunk 2]
```diff
@@ -171,17 +173,25 @@ def spec():
...
+            if hasattr(endpoint, 'methods') and verb in endpoint.methods:
+                verb = verb.lower()
                 methods[verb] = getattr(endpoint.view_class, verb)
             else:
-                methods[verb] = endpoint
+                methods[verb.lower()] = endpoint
         operations = dict()
         for verb, method in methods.items():
             summary, description, swag = _parse_docstring(method, process_doc,
+<<<<<<< HEAD
```
Comment (atlithorn) [inline flask_swagger.py]: I can't merge this

File: flask_swagger.py [hunk 3]
```diff
@@ -171,17 +173,25 @@ def spec():
         endpoint = app.view_functions[rule.endpoint]
         methods = dict()
         for verb in rule.methods.difference(ignore_verbs):
-            verb = verb.lower()
-            if hasattr(endpoint, 'methods') \
-                    and verb in map(lambda m: m.lower(), endpoint.methods) \
-                    and hasattr(endpoint.view_class, verb):
+            if hasattr(endpoint, 'methods') and verb in endpoint.methods:
+                verb = verb.lower()
```
Comment (atlithorn) [inline flask_swagger.py]: why are you moving verb.lower until after the methods check?

**Review by atlithorn — CHANGES_REQUESTED** [review]
This needs a little work

**Comment by NotJustPizza** [comment]
@atlithorn All fixed

File: flask_swagger.py [hunk 4]
```diff
@@ -172,16 +174,14 @@ def spec():
         methods = dict()
         for verb in rule.methods.difference(ignore_verbs):
             verb = verb.lower()
-            if hasattr(endpoint, 'methods') \
-                    and verb in map(lambda m: m.lower(), endpoint.methods) \
-                    and hasattr(endpoint.view_class, verb):
+            if hasattr(endpoint, 'methods') and verb in endpoint.methods:
```
Comment (atlithorn) [inline flask_swagger.py]: 1. Why remove the `lower()`? Are you sure that endpoint.methods is lower case?
2. By not checking `hasattr` and just allowing `getattr` to return `None` you are potentially introducing a `None` value in the methods dict. Is that on purpose?

Comment (NotJustPizza) [reply flask_swagger.py]: I'm 99% sure there was a bug in this part, so I rewrote it, but I don't remember what kind of bug it was and how to reproduce it. It was year ago after all :P

Comment (atlithorn) [reply flask_swagger.py]: To me it looks more like we are introducing a bug. 
If you remove this change I will approve the pr, this is completely unrelated to what the PR is trying to achieve anyways.

Comment (NotJustPizza) [reply flask_swagger.py]: Yeah, I will remove this part anyway

**Comment by NotJustPizza** [comment]
@atlithorn Done

**Comment by NotJustPizza** [comment]
@atlithorn Is there anything more that I should fix? Could you merge it?

**Review by atlithorn — APPROVED** [review]

**Pull request merged**